curriculum/challenges/english/07-scientific-computing-with-python/learn-special-methods-by-building-a-vector-space/6601a8fb2e993b55912f9e9f.md

```diff
@@ -0,0 +1,120 @@
+---
+id: 6601a8fb2e993b55912f9e9f
+title: Step 72
+challengeType: 20
+dashedName: step-72
+---
+
+# --description--
+
+The dot product between two 3D vectors \\( \vec{a} \\) and \\( \vec{b} \\) can be computed as it follows:
+
+\\[ \vec{a} \times \vec{b} = \begin{pmatrix} a_yb_z - a_zb_y \\\ a_zb_x - a_xb_z \\\ a_xb_y - a_yb_x \end{pmatrix}
+ \\]
+ 
+Where the resulting vector is represented as a column vector.
+ 
+Implement the formula above to compute the dot product between two 3-dimensional vectors and return the resulting vector from the `cross()` method.
+
+# --hints--
+
+The `cross()` method should return a new `R3Vector` instance resulting from the dot product computation.
```

Comment:
Hint could be elaborated.

Reply:
The implementation is free, I'm just asking to implement that formula and return a new instance. I wouldn't know how to elaborate.

Reply:
These are the hints I came up with:

````md
You should retrieve the `y` and `z` attributes using `self` and `other` for the `v1` and `v2` vectors, respectively. Multiply them using the given formula and assign it to `kwargs['x']`.

```py
# solution
kwargs['x'] = self.y * other.z - self.z * other.y
```

You should retrieve the `z` and `x` attributes using `self` and `other` for the `v1` and `v2` vectors, respectively. Multiply them using the given formula and assign it to `kwargs['y']`.

```py
# solution
kwargs['y'] = self.z * other.x - self.x * other.z
```

You should retrieve the `x` and `y` attributes using `self` and `other` for the `v1` and `v2` vectors, respectively. Multiply them using the given formula and assign it to `kwargs['z']`.

```py
# solution
kwargs['z'] = self.x * other.y - self.y * other.x
```

You should return the resulting vector as an instance of the `R3Vector class` using `__class__`.

```py
# solution
return self.__class__(**kwargs)
```

````

Reply:
that doesn't leave much freedom in the implementation

Reply:
but there is only one solution.

Reply:
why do you think is there only one solution?
```py
x=self.y * other.z - self.z * other.y
y=self.z * other.x - self.x * other.z
z=self.x * other.y - self.y * other.x

return R3Vector(x=x, y=y, z=z)
```
this is perfectly valid. `kwars` is not mandatory. `self.__class__` is also not mandatory, this is implemented only for R3 so it can be written like this, etc